tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -91,6 +90,102 @@ int main(int argc, char **argv)
         };
     };
 
+    /* s2n_security_policy_validate_certificate_chain */
+    {
+        int valid_sig_nid = s2n_ecdsa_sha256.libcrypto_nid;
+        int valid_hash_nid = 0;
+        EXPECT_SUCCESS(s2n_hash_NID_type(s2n_ecdsa_sha256.hash_alg, &valid_hash_nid));
+
+        int invalid_sig_nid = s2n_rsa_pkcs1_sha256.libcrypto_nid;
+        int invalid_hash_nid = 0;
+        EXPECT_SUCCESS(s2n_hash_NID_type(s2n_rsa_pkcs1_sha256.hash_alg, &valid_hash_nid));
+
+        struct s2n_cert_info valid = {
+            .self_signed = false,
+            .signature_nid = valid_sig_nid,
+            .signature_digest_nid = valid_hash_nid,
+        };
+        struct s2n_cert root = { 0 };
+        root.info = valid;
+        root.info.self_signed = true;
+
+        struct s2n_cert intermediate = { 0 };
+        intermediate.info = valid;
+        intermediate.next = &root;
+
+        struct s2n_cert leaf = { 0 };
+        leaf.info = valid;
+        leaf.next = &intermediate;
+
+        struct s2n_cert_chain cert_chain = { 0 };
+        cert_chain.head = &leaf;
+
+        struct s2n_cert_chain_and_key chain = { 0 };
+        chain.cert_chain = &cert_chain;
+        
+        /* valid chain */
+        {
+            EXPECT_OK(s2n_security_policy_validate_certificate_chain(&test_sp, &chain));
+        };
+
+        /* an invalid root signature is ignored */
+        {
+            root.info.signature_nid = invalid_sig_nid;
+            root.info.signature_digest_nid = invalid_sig_nid;
+            EXPECT_OK(s2n_security_policy_validate_certificate_chain(&test_sp, &chain));
+        };
+
+        /* an invalid intermediate causes a failure */
+        {
+            intermediate.info.signature_nid = invalid_sig_nid;
+            intermediate.info.signature_digest_nid = invalid_sig_nid;
+            EXPECT_ERROR_WITH_ERRNO(
+                    s2n_security_policy_validate_certificate_chain(&test_sp, &chain),
+                    S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT);
+        }
+    };
+
+    /* s2n_config_set_cipher_preferences */
+    {
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *invalid_cert = NULL,
+                s2n_cert_chain_and_key_ptr_free);
+        EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&invalid_cert, "rsae", "pss", "4096", "sha384"));
+
+        /* when certificate preferences apply locally and a config contains invalid
+         * certs then s2n_config_set_cipher_preferences fails
+         */
+        {
+            /* test assumptions */
```

Comment:
Yeah I also don't think these test assumptions add anything. These conditions are implied by the output of the test you wrote.